ggshield/verticals/secret/secret_scan_collection.py

```diff
@@ -21,19 +21,31 @@ class Result(NamedTuple):
     # TODO: Rename `file` to `scannable`?
     file: Scannable  # filename that was scanned
     scan: ScanResult  # Result of content scan
+    is_on_patch: bool  # Is the scan on a patch
+
+    def __init__(self, file: Scannable, scan: ScanResult):
+        self.file = file
+        self.scan = scan
+        self.is_on_patch = file.filemode != Filemode.FILE
```

Comment:
This can be a property, then Result can remain a NamedTuple, which implements `__eq__()` for you.